addon/adapters/osf-adapter.js

```diff
@@ -18,97 +24,199 @@ export default DS.JSONAPIAdapter.extend(HasManyQuery.RESTAdapterMixin, DataAdapt
         // Fix issue where CORS request failed on 301s: Ember does not seem to append trailing
         // slash to URLs for single documents, but DRF redirects to force a trailing slash
         var url = this._super(...arguments);
+        var options = (snapshot ? snapshot.adapterOptions : false) || {};
         if (requestType === 'deleteRecord' || requestType === 'updateRecord' || requestType === 'findRecord') {
             if (snapshot.record.get('links.self')) {
                 url = snapshot.record.get('links.self');
             }
+        } else if (options.url) {
+            url = options.url;
         }
+
         if (url.lastIndexOf('/') !== url.length - 1) {
             url += '/';
         }
         return url;
     },
     /**
-     * Construct a URL for a relationship create/update/delete. Has the same
-     * signature as buildURL, with the addition of a 'relationship' param
+     * Construct a URL for a relationship create/update/delete.
      *
      * @method _buildRelationshipURL
+     * @param {DS.Snapshot} snapshot
      * @param {String} relationship the relationship to build a url for
      * @return {String} a URL
      **/
-    _buildRelationshipURL(modelName, id, snapshot, requestType, query, relationship) { // jshint ignore:line
+    _buildRelationshipURL(snapshot, relationship) {
         var links = relationship ? snapshot.record.get(
             `relationshipLinks.${Ember.String.underscore(relationship)}.links`
         ) : false;
         if (links && (links.self || links.related)) {
             return links.self ? links.self.href : links.related.href;
+        }
+        return null;
+    },
+    _createRelated(store, snapshot, createdSnapshots, relationship, url, isBulk = false) { // jshint ignore:line
```

Comment:
As a totally general remark, I'm encouraging everyone (including myself) to add YUIDoc comments to new functions, documenting their intended purpose and arguments (or at least any unique ones).